server/container/guice/guice-common/src/main/java/org/apache/james/utils/ConfigurationsPerformer.java

```diff
@@ -20,22 +20,46 @@
 package org.apache.james.utils;
 
 import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.james.lifecycle.api.Configurable;
 
 import com.google.inject.Inject;
 
 public class ConfigurationsPerformer {
 
     private final Set<ConfigurationPerformer> configurationPerformers;
+    private final Configurables configurables;
 
     @Inject
-    public ConfigurationsPerformer(Set<ConfigurationPerformer> configurationPerformers) throws Exception {
+    public ConfigurationsPerformer(Set<ConfigurationPerformer> configurationPerformers, Configurables configurables) {
         this.configurationPerformers = configurationPerformers;
+        this.configurables = configurables;
     }
 
     public void initModules() throws Exception {
-        for(ConfigurationPerformer configurationPerformer : configurationPerformers) {
-            configurationPerformer.initModule();
-        }
+        
+        Set<ConfigurationPerformer> processed = processConfigurables();
+        
+        processOthers(processed);
```

Comment:
Instead of collecting and the processing others, a simple function with a if statement would be clearer and need only one loop run, no?